.github/workflows/e2e-tests.yml

```diff
@@ -12,20 +12,185 @@ on:
         type: string
 
 jobs:
+  setup:
+    name: Set up dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+            node-version: lts/*
+
+      # Check Node deps from previous workflow runs.
+      # Cache hit as long as package-lock.json file hasn't changed.
+      # Skip e2e-setup-ci step on cache hit.
+      - name: Cache Node.js dependencies
+        id: cache-node
+        uses: actions/cache@v4
+        with:
+            path: e2e/node_modules
+            key: ${{ runner.os }}-node-${{ hashFiles('e2e/package-lock.json') }}
+            restore-keys: |
+              ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        if: steps.cache-node.outputs.cache-hit != 'true'
+        run: make e2e-setup-ci
+
+      # Tool allowing for advanced Docker builds, including caching
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      #
+      - name: Build Docker image with GitHub Actions cache
+        uses: docker/build-push-action@v5
+        with:
+          # Directory containing all files Docker needs for build
+          context: .
+          # Path to relevant Dockerfile
+          file: ./e2e/Dockerfile
+          # Whether to push to a Docker registry
+          push: false
+          # Tag for the image buing build
+          tags: playwright-e2e:latest
+          # Where to pull cache layers from - gha means it will pull from Github Actions cache
+          cache-from: type=gha
+          # Where to store build cache after the build is complete
+          # mode max specifies to cache all layers
+          cache-to: type=gha,mode=max
+          # Load the built image from Docker Buildx onto the local Docker daemon
+          load: true
+
+      - name: Save Docker image to tar file
+        run: |
+            docker save playwright-e2e:latest -o /home/runner/work/_temp/playwright-e2e.tar
+
+      - name: Upload Docker image artifact
+        uses: actions/upload-artifact@v4
+        with:
+            name: playwright-e2e
+            path: /home/runner/work/_temp/playwright-e2e.tar
+            retention-days: 1
+
```

Comment:
Cache Docker layers using `buildx` and upload to artifact

Reply:
cc @acouch @lorenyu  - this buildx caching and then upload to github actions artifact actually ends up taking longer then just building the image with `make e2e-test` in each shard - so will probably get rid of this.